### DISCUSSION TITLE
LIS Run Quits When Writing Streamflow Output (LISF v7.4, Noah-MP v4.0.1, HYMAP2)

### DISCUSSION THREAD

**Discussion by osbornesci** [NASA-LIS/LISF#1109] [Q&A] [answered]
Greetings - I'm running into an error message within the lislog related to the streamflow settings in the model output attributes table:
```
[INFO] Noah-MP.4.0.1 archive restart written:
 ./OUTPUT/adapted_amazon_hymap2_new/SURFACEMODEL/200105/LIS_RST_NOAHMP401_200105
 210000.d01.nc
 [ERR] Streamflow                                         field is not defined
 [ERR] for diagnostic output...
 [ERR] Please exclude it from the model output attributes table
 [ERR] Program stopping ..
 [ERR] endrun is being called
```
In a previous LIS test (identical setup except with a different value for streamflow within the output .TBL - please see below), the lislog reported a similar streamflow output formatting error, so I tried updating the streamflow routing setting within the MODEL_OUTPUT_LIST.TBL based on information from a recent LIS discussion (https://github.com/NASA-LIS/LISF/discussions/857). 

Updated line -- `Streamflow:     1 m3/s  -  1 0 0 1 333 10`      # Streamflow (prev: 256 instead of 333)

I'm attaching the ldt and lis config files, the output attributes table, ldtlog file, and lislog files (4 processors) here:
[osborne_sharefiles.zip](https://github.com/NASA-LIS/LISF/files/8848655/osborne_sharefiles.zip)

When I do _not_ activate HyMAP routing, LIS successfully runs to completion.

Is there a certain edit I can make to allow LIS to run successfully when HyMAP2 is activated? Possibly something related to the MODEL_OUTPUT_LIST.TBL settings for streamflow or HyMAP2-related variables?

I appreciate your suggestions. Thank you.

**Comment by zaitchik** [comment]
Just checking in on the question above. Any help on OUTPUT.TBL entries or other suggestions to get HyMAP2 output would be appreciated!

**Comment by karsenau** [comment]
Hi @osbornesci and @zaitchik:

Thanks for sharing this situation that you are experiencing.  Based on the information, log and config files provided, I can't tell from the information why it would crash at that spot, as your entries look fine. 

I might recommend reducing a lot of the entries in your MODEL_OUTPUT_LIST.TBL and only including the ones that you would need. I typically start with just a handful of relevant fields, e.g., water balance and energy balance terms, some input forcing fields, etc. See if removing a lot of MODEL output fields just to some basic ones, and only include Streamflow: and no other HyMAP2 output variables in your tests.  I don't see any conflicts with how you are turning on HYMAP2 for the router and turning on Streamflow in your output table. Units and other selected options look fine in your output table too.

Try to test with less output fields and feel free to share if that helps.  If not, we can try to ping one of our team members very familiar with the code.

Thanks!

**Reply by osbornesci** [reply]
Hi @karsenau:

Thank you kindly for looking into this, and for your suggestions! 

I'm now trying a LIS run on Discover after having reduced the number of variables included within the MODEL_OUTPUT_LIST.TBL (and only including streamflow as a HyMAP2 variable). I will report here on the results of this run.

Ben and I have also been in contact with @agetiran via email (thank you!). Augusto ran LIS on his profile with the MODEL_OUTPUT_LIST.TBL as is (from above in post), and reported that this file did not present a problem for him either. 

He suggested we try updating the HyMAP section of MODEL_OUTPUT_LIST.TBL to match the one he uses for HyMAP2 runs (using 333 instead of 256 for all HyMAP variables, plus a few other modifications compared to the version we initially used). Here is the new section we used: 
<img width="934" alt="OutputTableSettings_HyMAPsection" src="https://user-images.githubusercontent.com/99269620/178068794-58a7626d-97a8-40aa-ab24-607e33affe1e.png">

After running LIS with this update to the HyMAP section of the MODEL_OUTPUT_LIST.TBL, we received a similar error message with the streamflow variable, here: 
<img width="627" alt="ErrorMessage" src="https://user-images.githubusercontent.com/99269620/178068950-76dd0f28-c71a-4ea4-b52d-53dee8785a7b.png">

`Streamflow field is not defined for diagnostic output. Please exclude it from the model output attributes table`

We wonder if the errors could have anything to do with my specific Discover profile environment / settings or something that might have affected the LIS v7.4 executable file during configuration / compilation on my profile. Please let us know if you have additional suggestions moving forward, and we'll be happy to try them to work towards successful runs. I will report back on the current LIS run that includes a reduced number of output variables. 

Thank you again,
Tashiana

**Reply by osbornesci** [reply]
Hi @karsenau,

I'm following up here on the most recent LIS runs I tried after following your suggestion (thank you) to reduce many of the entries in the MODEL_OUTPUT_LIST.TBL. 

I ran LIS with about half or fewer or the MODEL_OUTPUT_LIST.TBL entries commented out, and also with every entry except for the streamflow entry commented out. Both returned the same "streamflow field is not defined" error as previously (image below).

<img width="628" alt="Screen Shot 2022-07-12 at 10 31 58 AM" src="https://user-images.githubusercontent.com/99269620/178621084-3fa9291d-3267-46db-89b3-5b5e2ce439a1.png">

We're glad to keep trying additional troubleshooting steps if there's something you or your team recommends moving forward. Thank you!

**Reply by karsenau** [reply]
Hi @osbornesci:

I have looked through your files but still not quite sure why you would get this error.  The only other thing that I could think of is what you have set in your "user.cfg" file in the make directory, if you have that set.  I see in your lislog file that HyMAP is being set to run but I'm not sure if there is some issue in your compilation of all HyMAP routines?

One recommendation I would have:  Check out the latest version of our LIS7.4 public branch, configure and compile (with default settings) with the latest suggested private module file with the Intel compiler (if you are using Intel) in the "env/discover folder, which would be  lisf_7_intel_2021.4.0, on Discover.
Then try running with the current LIS config file and MODEL output table that you have (and already had checked by @agetiran).

That should work and not give any real issue.

Hope that helps!

**Reply by osbornesci** [reply]
Hi @karsenau:

Thank you so much for looking through our setup in such detail. We truly appreciate your help!

In addition to stepping through your great suggestions above, I will also try a test case that @agetiran shared that uses HYMAP2. Thank you both again!

**Comment by PrateekSharma99** [comment]
I am facing a somewhat similar issue but in my case its SoilMoist that shows the error while writing the .netcdf files as output. 
![Screenshot (34)](https://user-images.githubusercontent.com/89891132/215719831-be4450fb-0712-4ec2-a8e1-76003c506fc3.png)

**Reply by PrateekSharma99** [reply]
The model works fine if I exclude the SoilMoist is excluded but that's not the suitable option for me

**Reply by PrateekSharma99** [reply]
Any suggestions on this would be welcome PFA the output tbl file and the lis.config file
[Prateel.zip](https://github.com/NASA-LIS/LISF/files/10553411/Prateel.zip)

**Reply by prakrutkansara** [reply]
Hi @Daduji123! I faced a similar error and it is related to the units of soil moisture inside the MODEL_OUTPUT_LIST.TBL. Try changing the soil moisture information inside your MODEL_OUTPUT_LIST.TBL with the following:

SoilMoist:    1  m3/m3   -    0 0 0 4  86 1000    # Average layer soil moisture (m3/m3)

**Reply by sudarsan3010** [reply]
@prakrutkansara Thanks a lot. I faced similar issue and now rectified.

**Comment by zhengzhenk** [comment]
Hi @osbornesci , have you resolved the issue you encountered?

I'm also using LISF v7.4 and Noah-MP v4.0.1 to run the HYMAP2 routing module. Regardless of whether I set the streamflow routing in MODEL_OUTPUT_LIST.TBL as follows:
`Streamflow: 1 m3/s - 1 0 0 1 333 10`
or
`Streamflow: 1 m3/s - 1 0 0 1 256 10`,
the HYMAP2 routing module seems to run fine.

However, in this case, I'm using HYMAP2 routing parameters from the LIS data portal, provided as 0.25° resolution parameter files (LIS data portal website is here):
`https://portal.nccs.nasa.gov/lisdata_pub/data/PARAMETERS/HYMAP_parms/`
This resolution is relatively coarse for my modeling purposes, and I'd like to test with some finer resolution parameters. I noticed in your attachments that you used HYMAP2 routing parameters with a 0.1° resolution. Could you please tell me how to create these parameters or where to download them? Any advice would be greatly appreciated.

Thank you!

**Reply by dmocko** [reply, marked as answer]
Hi @zhengzhenk 

I am closing this question, as it's unrelated to the original question posted on this discussion.

I will answer your issue on Discussion #1444
